src/backend/marsha/bbb/tests/api/classroom/test_invite.py

```diff
@@ -0,0 +1,90 @@
+"""Tests for the classroom invite API."""
+import secrets
+
+from django.test import TestCase, override_settings
+
+from marsha.bbb.factories import ClassroomFactory
+from marsha.core.models import INSTRUCTOR, NONE
+from marsha.core.simple_jwt.tokens import ResourceAccessToken
+from marsha.core.tests.testing_utils import reload_urlconf
+
+
+# We don't enforce arguments documentation in tests
+# pylint: disable=unused-argument
+
+
+@override_settings(BBB_API_ENDPOINT="https://10.7.7.1/bigbluebutton/api")
+@override_settings(BBB_API_SECRET="SuperSecret")
+@override_settings(BBB_ENABLED=True)
+class ClassroomInviteAPITest(TestCase):
+    """Test classroom invite endpoint."""
+
+    maxDiff = None
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # Force URLs reload to use BBB_ENABLED
+        reload_urlconf()
+
+    def test_missing_token(self):
+        """Missing invite token in the query string should return a 400."""
+
+        classroom = ClassroomFactory()
+
+        response = self.client.get(f"/api/classrooms/{classroom.id}/invite/")
+
+        self.assertEqual(response.status_code, 400)
+        content = response.json()
+        self.assertEqual(content, {"message": "missing required token in query string"})
+
+    def test_invite_public_token(self):
+        """Using a public token should return a resource token with no permission."""
+
+        classroom = ClassroomFactory()
+
+        response = self.client.get(
+            f"/api/classrooms/{classroom.id}/invite/?token={classroom.public_token}"
```

Comment:
I would like to question the choice of the name "invite" for this endpoint. I understood that this endpoint is called to invite a user to the classroom. In fact it is called to retrieve a token. Maybe call it "token"?

Reply:
in that case the query string can be rename from `token` to `invite` or `invite_token` ?